spek-runner/junit5/src/main/kotlin/org/spekframework/spek2/junit/SpekTestDescriptor.kt

```diff
@@ -71,8 +74,19 @@ class SpekTestDescriptor internal constructor(
 
     override fun getTags(): MutableSet<TestTag> = mutableSetOf()
 
-    override fun removeFromHierarchy() = throw UnsupportedOperationException()
-    override fun removeChild(descriptor: TestDescriptor) = throw UnsupportedOperationException()
+    override fun removeFromHierarchy() {
+        parent.ifPresent { parent ->
+            parent.removeChild(this)
+        }
+    }
+
+    override fun removeChild(descriptor: TestDescriptor) {
+        if (scope is GroupScopeImpl) {
```

Comment:
Removal relies on instance equality, each unique scope will only have one instance throughout the lifetime of the runtime.